### ISSUE TITLE
feat(DatePicker): Month range selection in mat-datepicker / mat-calendar

### ISSUE THREAD

**Issue by mit-bhalodi** [angular/components#27922]
### Feature Description

Proposed Implementation:
I suggest adding an option to the **mat-date-picker/mat-calendar** component that allows users to select a range of dates by clicking on a start date and an end date within the same calendar view. This would provide a more intuitive and efficient way to handle month-based selections. the similar implementation is observed in **mat-date-range-picker**

### Use Case

The month range selection feature would be particularly useful in scenarios where users need to filter or select data over a specific period, such as monthly reports, statistics, or date-based filtering. Eg. salary reports, time and attendance reports etc

**Comment by andrewseguin** [comment]
Can you clarify more about what you mean about it being more efficient to handle month-based selections if there was an option to select ranges in mat-datepicker?

**Comment by stephan-strate** [comment]
Any news on this topic? I would say its not only a convenience feature, sometimes its necessary to have such an input.

**Comment by mit-bhalodi** [comment]
Hello team, any updates on this feature requirement?

**Comment by forgantini** [comment]
any news?

**Comment by dwoodv2** [comment]
> Any news on this topic? I would say its not only a convenience feature, sometimes its necessary to have such an input.

Yeah this would be a very useful feature, [PrimeNG has this functionality](https://primeng.org/datepicker#monthpicker) and it works with months, years etc. out of the box, so I would love this to be added to Angular Material at some point
